### PR TITLE
Add Android Signer Sample

### PR DESCRIPTION
**Pull request by tobiasheine**
### Problem
This adds an Android sample for the signer library.

### Considerations
- Add android example project
- consume library as gradle composite build
- change artifact id to `com.novoda:aws-v4-signer`

### Paired with
@mr-archano on the composite build and gradle module structure part 🙏

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -8,8 +8,7 @@ pluginManagement {
     }
 }
 
-rootProject.name = 'aws-v4-signer'
-
-include ':library'
+include ':aws-v4-signer'
```
Comment (tobiasheine) [inline settings.gradle:11]: This renames the `library` module to `aws-v4-signer`.

File: samples/android/settings.gradle [hunk 2]
```diff
@@ -0,0 +1,8 @@
+rootProject.name = 'aws-v4-signer-android-sample'
+include ':app'
+
+includeBuild('../../') {
+    dependencySubstitution {
+        substitute module('com.novoda:aws-v4-signer-jvm') with project(':aws-v4-signer')
```
Comment (tobiasheine) [inline samples/android/settings.gradle]: This includes the library to the sample as composite build.

File: samples/android/app/build.gradle [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation"org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlin_version"
+    implementation 'androidx.appcompat:appcompat:1.1.0-alpha05'
+    implementation 'androidx.core:core-ktx:1.2.0-alpha01'
+    implementation 'androidx.constraintlayout:constraintlayout:1.1.3'
+    implementation 'com.google.android.material:material:1.1.0-alpha06'
+    implementation 'com.novoda:aws-v4-signer-jvm:local'
```
Comment (tobiasheine) [inline samples/android/app/build.gradle]: Here we add the signer as a dependency to the sample.

File: samples/android/app/build.gradle [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    implementation 'com.squareup.okhttp3:okhttp:4.0.0-alpha02'
```
Comment (juankysoriano) [inline samples/android/app/build.gradle]: What uses okhttp? I couldn't find what's using it.

If nothing is using it then I'd say let's remove it as can cause confusion

**Review by juankysoriano — APPROVED** [review]

**Pull request merged**